### PR TITLE
Update CompositeAggregationBuilder.java

### PR DESCRIPTION
**Pull request by FabioMNAraujo**
We want to fix this issue for a university project and we think these modifications are leading us in the right direction, but we need some help and more time to analyze more code.
This relates to: https://github.com/elastic/elasticsearch/issues/28611?fbclid=IwAR3kGEH9maEtAB--lDn6NUyHYpUQMrJCqaShQUkvNDnDYvcsBqZ9xtcuTBA

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregationBuilder.java [hunk 1]
```diff
@@ -154,9 +154,9 @@ public int size() {
     @Override
     protected AggregatorFactory<?> doBuild(SearchContext context, AggregatorFactory<?> parent,
                                            AggregatorFactories.Builder subfactoriesBuilder) throws IOException {
-        if (parent != null) {
+        /*if (parent != null) {
             throw new IllegalArgumentException("[composite] aggregation cannot be used with a parent aggregation");
-        }
+        }*/
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregationBuilder.java]: We cannot accept any parent here so as explained in https://github.com/elastic/elasticsearch/issues/28611 we should check that all parent are instances of NestedAggregatorFactory. Something like:
````
private boolean checkParentIsNullOrNested(AggregatorFactory<?> parent) {
        if (parent == null) {
            return true;
        } else if (parent instanceof ReverseNestedAggregatorFactory) {
            return checkParentIsNullOrNested(parent.getParent());
        } else {
            return false;
        }
    }
````
Then we need a test that checks that nested aggregation and composite aggregation works fine together.  `org.elasticsearch.search.aggregations.bucket.NestedIT` would be a good place to write such tests.

**Review by jimczi — CHANGES_REQUESTED** [review]

**Comment by FabioMNAraujo** [comment]
> Pinging @elastic/es-analytics-geo

 i didn't know how to call the "doBuild" function. I created in the test a CompositeBuilder and a NestedBuilder and sub-aggregated the Composite to a Nested but I don't think the test I've created actually tests the changes added to the code. If anyone could help improve the testing, I'd be grateful

**Comment by javanna** [comment]
@jimczi would you be able to help here? Thanks!

**Comment by polyfractal** [comment]
I think this was closed by https://github.com/elastic/elasticsearch/pull/37178, please feel free to re-open if that's not true or I misunderstood :)

**Pull request closed without merging**